tests/system_tests/__init__.py

```diff
@@ -146,23 +146,23 @@ def lifecycle_handshake(session, command='launch', options=None,
                         breakpoints=None, excbreakpoints=None,
                         threads=False):
     with session.wait_for_event('initialized'):
-        req_initialize = session.send_request(
+        req_initialize = session.send_request_and_wait(
             'initialize',
             adapterID='spam',
         )
-    req_command = session.send_request(command, **options or {})
-    req_threads = session.send_request('threads') if threads else None
+    req_command = session.send_request_and_wait(command, **options or {})
+    req_threads = session.send_request_and_wait('threads') if threads else None
 
     reqs_bps = []
     reqs_exc = []
     for req in breakpoints or ():
         reqs_bps.append(
-            session.send_request('setBreakpoints', **req))
+            session.send_request_and_wait('setBreakpoints', **req))
```

Comment:
Why are we changing the behavior to be synchronous?
System tests should replicate real world, send multiple requests and receive then as they arrive, instead of send one request, wait for that to be processed. 
If we don't do that, I personally see no value in system tests.